modified to use sdl2 for an android install

snuq opened the pull request:
I managed to get audiostream compiled and working using sdl2 on android.
Mostly just changed instances of 'SDL' into 'SDL2' and added an include.
To be viable tho, this needs three things:
1. able to use sdl or sdl2 - I do not know how to detect which is being used to add if/else clauses
2. make the include system-independent - I'm not sure how to refer to that directory from environmental variables, I'm not sure how to get the ndk path.
3. the python-for-android init file needs to be edited as well, as in: https://github.com/kivy/python-for-android/pull/1178

KeyWeeUsr commented on setup.py:
```diff
@@ -67,15 +67,15 @@ def makeExtension(extName, files=None):
     include_dirs = [
             join(kivy_ios_root, 'build', 'include'),
-            join(kivy_ios_root, 'build', 'include', 'SDL')]
+            join(kivy_ios_root, 'build', 'include', 'SDL2')]
```
I'm not sure about this one, but it might be even a good fix. Kivy-ios recipe has `SDL2`, however there it's just `SDL`. I wonder how does it even compile for iOS with mixed libs. @akshayaurora @Zen-CODE ?

Zen-CODE replied:
Yes, agreed. These changes seem specifically targeted at Android, so changes for other OS's should be avoided.@snuq. Have you tested these changes on Mac/iOS?

snuq replied:
I did not test it, no, i guess i got carried away changing all the instances of SDL

KeyWeeUsr commented on setup.py:
```diff
@@ -67,15 +67,15 @@ def makeExtension(extName, files=None):
 
 elif platform == "darwin":
-    include_dirs.append('/usr/local/include/SDL')
+    include_dirs.append('/usr/local/include/SDL2')
```
This is macOS and you'll break the `SDL`-only compatibility. I think there should be `USE_SDL2` env var to check before choosing which `SDL` version to choose i.e.:

`USE_SDL2=0` → `/usr/local/include/SDL`
`USE_SDL2=1` → `/usr/local/include/SDL2`

See the more global-ish approach in [Kivy `setup.py`](https://github.com/kivy/kivy/blob/0456eff89bccf93fa6bd8aada318246082f1b8cc/setup.py#L150), but basically it's about this:

    use_sdl2 = os.environ.get('USE_SDL2')  # value or None
    include_dirs.append('/usr/local/include/SDL2' if use_sdl2 else '/usr/local/include/SDL')
    libraries = ['SDL2', 'SDL2_mixer'] if use_sdl2 else ['SDL', 'SDL_mixer']
    etc

KeyWeeUsr commented on setup.py:
```diff
@@ -30,19 +30,19 @@
 
 if not have_cython:
-    libraries = ['sdl', 'sdl_mixer']
+    libraries = ['SDL2', 'SDL2_mixer']
```
Same with the `USE_SDL2` flag as in the other comment.

KeyWeeUsr commented on setup.py:
```diff
@@ -30,19 +30,19 @@
 else:
     include_dirs.append('.')
-    include_dirs.append('/usr/include/SDL')
+    include_dirs.append('/usr/include/SDL2')
```
It seems to me that this part is linux only (I might be wrong and it's macos too), so I'd probably move it down to the conditions under either `elif platform == 'linux'` or rather just under `else` because it's `sys.platform`. Otherwise I think we end up with this state anyway e.g. if on iOS:

`include_dirs`:

    [
        '/usr/include/SDL2',  # most likely unused
        '/home/kivy/.buildozer/android/platform/android-ndk-r9c/sources/android/support/include',  # bad
        '/usr/include/SDL2',  # duplicate
        join(kivy_ios_root, 'build', 'include'),  # actually used include
        join(kivy_ios_root, 'build', 'include', 'SDL2'),  # actually used include
    ]

KeyWeeUsr commented on setup.py:
```diff
@@ -30,19 +30,19 @@
 
 # configure the env
-libraries = ['SDL', 'SDL_mixer']
+libraries = ['SDL2', 'SDL2_mixer']
```
Same with the `USE_SDL2` flag as in the other comment.

inclement commented:
@Zen-CODE @KeyWeeUsr Just to be clear, I asked @snuq to open this PR as the first step towards merging the changes, knowing that they'd need some fixes like this to avoid interfering with other OSs.

@snuq Do the changes suggested make sense to you?

snuq commented:
Most of the suggestions made sense, yes, I made some changes to the file, added the environmental variable USE_SDL2, and removed the changes for ios and osx.  Ill have to dig for the ndk environ variable once i get some more free time.

KeyWeeUsr commented on setup.py:
```diff
@@ -30,19 +30,19 @@
+    libraries = ['SDL2', 'SDL2_mixer'] if use_sdl2 else ['sdl', 'sdl_mixer']
 else:
-    include_dirs.append('.')
-    include_dirs.append('/usr/include/SDL')
```
This might cause troubles as it's not meant for android (or iOS).

KeyWeeUsr commented on setup.py:
```diff
@@ -30,19 +30,19 @@
+libraries = ['SDL2', 'SDL2_mixer'] if use_sdl2 else ['SDL', 'SDL_mixer']
 library_dirs = []
-include_dirs = ['/usr/include/SDL']
+include_dirs = ['/usr/include/SDL2', '/home/kivy/.buildozer/android/platform/android-ndk-r9c/sources/android/support/include'] if use_sdl2 else ['/usr/include/SDL2']
```
Same as before, use the NDK folder directly in P4A recipe, not here :) If you need, just write the raw path to the recipe and check with @inclement how to get the right path, however I think it's somewhere in the `self.ctx` part as I said before.

snuq commented:
ok... i really have no idea what im doing at this point... 
I have no idea if i should set the environmental variable 'SDL2_INCLUDE_DIR' for the include folder like I have, or if i should somehow put it in the CFLAGS environmental variable... 
on top of that, i have been looking for a reference to the ndk folder and still cant find one.
In its current state, it will compile and works, but I think im out of my league fixing this to be release-worthy...

KeyWeeUsr commented:
I'd say it's good enough, but there's still the `else` part you edited out and I think it might cause troubled (as mentioned in the review comment).

Also, this line:

```python
libraries = ['SDL2', 'SDL2_mixer'] if use_sdl2 else ['sdl', 'sdl_mixer']
```

might need the libraries to be lowercase as they were originally:

```python
libraries = ['sdl2', 'sdl2_mixer'] if use_sdl2 else ['sdl', 'sdl_mixer']
```

but I might be wrong. If it works with lowercase, no need to force caps though.

@inclement ?

snuq commented:
alright, i changed the includes back, the caps are needed tho, tested a compile with it lowercase and it failed.

KeyWeeUsr commented:
Ah, ok... then it seems fine to me. Thanks! :)
